torch/distributed/fsdp/_fully_shard/_fsdp_param_group.py

```diff
@@ -414,11 +414,14 @@ def post_backward(self, *unused: Any):
         if len(fsdp_params_with_grad) == 0:
             return
         with record_function(self._with_fqn("FSDP::post_backward_reduce")):
-            if self.comm_ctx.reduce_scatter_state is not None:
+            if (
+                self.comm_ctx.reduce_scatter_state is not None
+                and self.comm_ctx.reduce_scatter_state.event is not None
+            ):
                 self.device_handle.current_stream().wait_event(
                     self.comm_ctx.reduce_scatter_state.event
                 )
-                self.comm_ctx.reduce_scatter_state = None
+            self.comm_ctx.reduce_scatter_state = None
```

Comment:
nit: maybe it's better to keep the original style and do a double if statement like:
```
if self.comm_ctx.reduce_scatter_state is not None:
    if self.comm_ctx.reduce_scatter_state.event is not None:
        self.device_handle.current_stream().wait_event(
            self.comm_ctx.reduce_scatter_state.event
        )
    self.comm_ctx.reduce_scatter_state = None
```
there is semantic meaning to `if self.comm_ctx.reduce_scatter_state is not None:` -- namely that there was a preceding reduce-scatter that we have not waited on
if we ever want to refactor the stashing and waiting logic to be handled in a single API, it would be clearer if we only cleared when we need to